compiler/z/codegen/SystemLinkagezOS.cpp

```diff
@@ -615,15 +614,43 @@ TR::S390zOSSystemLinkage::genCallNOPAndDescriptor(TR::Instruction* cursor, TR::N
    {
    if (comp()->target().is32Bit())
       {
-      uint32_t callDescriptorValue = TR::XPLINKCallDescriptorSnippet::generateCallDescriptorValue(this, callNode);
-
-      TR::S390ConstantDataSnippet* callDescriptor = new (self()->trHeapMemory()) TR::XPLINKCallDescriptorSnippet(cg(), this, callDescriptorValue);
-      cg()->addDataConstantSnippet(callDescriptor);
+      // The XPLINK Call Descriptor is created only on 31-bit targets when:
+      // 
+      // 1. The call site is so far removed from the Entry Point Marker of the function that its offset cannot be contained
+      // in the space available in the call NOP descriptor following the call site.
+      // 
+      // 2. The call contains a return value or parameters that are passed in registers or in ways incompatible with non-
+      // XPLINK code.
+      // 
+      // The XPLINK Call Descriptor has the following format:
+      // 
+      //                                        0x01                               0x02                               0x03
+      // 0x00 +----------------------------------+----------------------------------+----------------------------------+----------------------------------+
+      //      | Signed offset, in bytes, to Entry Point Marker (if it exists)                                                                             |
+      // 0x04 +----------------------------------+----------------------------------+----------------------------------+----------------------------------+
+      //      | Linkage and Return Value Adjust  | Parameter Adjust                                                                                       |
+      //      +----------------------------------+----------------------------------+----------------------------------+----------------------------------+
+      //
+      // We generate the XPLINK call descriptor inline right after the call instead of the literal pool because some
+      // z/OS 31-bit programs tend to be rather large, and the distance between the call location and the literal pool
+      // may exceed the number of bits we have to encode the offset in the NOP descriptor.
+
+      TR::LabelSymbol* xplinkCallDesciptorBeginLabel = generateLabelSymbol(cg());
+      TR::LabelSymbol* xplinkCallDesciptorEndLabel = generateLabelSymbol(cg());
```

Comment:
Fix spelling of "Descriptor" in these variable names.  (unless that was intentional?)

Reply:
Fixed in [force-push](https://github.com/eclipse/omr/compare/aa4d2cf9e7c421a5ea39e9f8d701fd14e58ddbca..0b3376f0ba6f553545daf6908b91dc0a9729d50e).